torchtune/utils/metric_logging.py

```diff
@@ -317,3 +317,127 @@ def close(self) -> None:
         if self._writer:
             self._writer.close()
             self._writer = None
+
+
+class CometLogger(MetricLoggerInterface):
+    """Logger for use w/ Comet (https://www.comet.com/site/).
+    For more information about arguments expected by Comet, see
+    https://www.comet.com/docs/v2/guides/experiment-management/configure-sdk/#for-the-experiment.
+
+    Args:
+        api_key (Optional[str]): Comet API key. It's recommended to configure the API Key from the environment.
+        workspace (Optional[str]): Comet workspace name. If not provided, uses the default workspace.
+        project (Optional[str]): Comet project name. Defaults to Uncategorized.
+        experiment_name (Optional[str]): The name for comet experiment to be used for logging.
+        experiment_key (Optional[str]): The key for comet experiment to be used for logging. Must be an alphanumeric
+            string whose length is between 32 and 50 characters.
+        mode (Optional[str]): Control how the Comet experiment is started. "get": Continue logging to an existing
```

Comment:
you can make a bulleted list here
```suggestion
        mode (Optional[str]): Control how the Comet experiment is started. 
        
        - ``"get"``: Continue logging to an existing [...]
        - ``"create"``: [...]
        - ``"get_or_create"``: [...]
```
also minor nit, can we make the default option the first bullet?

Reply:
I give it a try but I cannot seems to have found the right format. Here is how it looks if I align it with the arguments list 
![Screenshot 2024-08-08 at 21-36-06 CometLogger — torchtune main documentation](https://github.com/user-attachments/assets/d5069827-6dae-4ab3-9a3b-7e3823034d98)

And I indent them even by one space, sphinx-build is complaining:

```
.../torchtune/torchtune/utils/metric_logging.py:docstring of torchtune.utils.metric_logging.CometLogger:20:Unexpected indentation.
```

Do you know what is the proper way to format it?

Reply:
I re-ordered the options but couldn't find the right format to render an indented list. I'm not sure if the Google docstring support it.